$\chi^2$ - test of _absolute_ frequencies (number of objects) in classifications

$\chi^2 = \Sigma\frac{(O-E)^2}{E}$

O = observed, E = expected

One frequency distribution

- Goodness-of-fit test

Two-way frequency distributions

- Cross tabulation

Fixed marginals

Homogeneity test

"No difference between A and B"

Random marginals

Independent Test

"The row classification is independent of the column classification"

Test of 3 hamburger brands

$H_0$: No difference in taste

$P(A) = \frac{1}{3}, P(B) = \frac{1}{3}, P(C) = \frac{1}{3}$

$H_1$: Difference in taste

In [10]:
def chi_square(obs, e):
    total = 0
    for o in obs:
        o_e = (o - e) * (o - e)
        total += (o_e / e)
    return total

Since the observed value of $\chi^2$ (8.4) is greater than the critical value (5.99) $H_0$ is rejected.

There is a difference between the hamburgers

$H_0$: equal frequency in the five age categories

$H_1$: Not $H_0$

$\alpha : 0.05$

$df = (K-1) = (5-1) = 4$

$\chi^2_{Table} = 9.49$

In [12]:
chi_square([6, 20, 35, 41, 48], 30)

38.2

In [13]:
def chi_square_more(l):
    sum = 0
    for i in l:
        sum += i
    exp = sum / len(l)
    return chi_square(l, exp)

In [16]:
freq = [152, 114, 106, 51, 43, 43]
pct = [30, 20, 20, 10, 10, 10]

total = 0
for f in freq:
    total += f

exp = [(total / 100 * x) for x in pct]

In [17]:
exp

[152.7, 101.8, 101.8, 50.9, 50.9, 50.9]

In [18]:
def chi_square_l(obs, exps):
    total = 0
    for o in zip(obs, exps):
        o_e = (o[0] - o[1]) * (o[0] - o[1])
        total += (o_e / o[1])
    return total

In [19]:
chi_square_l(freq, exp)

4.091028159790438

Since the observed value (4.09) is less than the critical value (11.)